src/cryptonote_protocol/cryptonote_protocol_handler.h

```diff
@@ -104,7 +104,7 @@ namespace cryptonote
     bool get_payload_sync_data(CORE_SYNC_DATA& hshd);
     bool on_callback(cryptonote_connection_context& context);
     t_core& get_core(){return m_core;}
-    bool is_synchronized(){return m_synchronized;}
+    virtual bool is_synchronized() const override {return !no_sync() && m_synchronized;}
```

Comment:
I didn't comment on this in the 0.17 review - it can be updated later (or just left as-is). I would mark the function as `final` too. The calls within the class can be de-virtualized+inlined easliy with this change.

Reply:
Actually, I think the other `virtual`+`override` functions in `cryptonote::core` can be marked as a `final` too.

Reply:
Updated both PRs